### ISSUE TITLE
[Bug] Rule variable-overwritten-before-usage should not be raised if overridden in an IF block

### ISSUE THREAD

**Issue by Leemur89** [MarketSquare/robotframework-robocop#950]
### What happened?

I am initiating a variable that I conditionally override in an IF block, but the warning variable-overwritten-before-usage is raised.

### What command/code did you try to run?

```robotframework
My Keyword
    [Documentation]  To reproduce the bug
    ${output}  Set Variable  ${TRUE}
    IF    os.path.isdir("/mydir")
        ${condition}  Set Variable  ${FALSE}  # here I have an actual keyword that performs an action an a check
        IF    $condition
            ${output}  Set Variable  ${FALSE}
        END
    END
    [Return]    ${output}
```

### What is the full error message?

Warning variable-overwritten-before-usage is raised

### What did you expect to happen instead?

Warning should not be raised if the variable is defined before the IF/ELSE block, and overridden in the IF or the ELSE
But if the variable is overridden in both IF and ELSE, then warning should be raised

### Robocop version

4.1.0

**Comment by bhirsz** [comment]
Thanks for the report. I was aware that this rule would trigger on such case but ruled it as fine for now. It can also happen with FOR or WHILE loops (for example when someone is defining default variable value in case the loop never iterates). Looking at it now I can indeed try to check if all if branches overwrite variable or not or variable is overwriten in the loop. I will check how much effort it will require

**Comment by rikerfi** [comment]
Really appreciate this fix as it generates plenty of noise :)

**Comment by bhirsz** [comment]
Released in 4.2.0
